### PR TITLE
Upload desktop integration artifacts to GCS

### PR DESCRIPTION
**Pull request by anonymous-akorn**
This creates a python script to upload artifacts to GCS, and adds a step to the integration test workflow to run it. 

We can't expose these artifacts in Github workflows, as that would expose secrets: in order to run a desktop integration test it's necessary to have the decrypted google-services.json file. This script thus uploads desktop artifacts (the integration test binaries, and the google services json) to GCS instead. 

To avoid duplication with the existing test_lab.py tool, common functionality has been moved into the new library file integration_testing/gcs.py

At the end of the workflow step to upload artifacts, it will report this:
Use 'gsutil cp <gs_uri> <local_path> to copy an artifact locally.
Use 'gsutil -m cp -r gs://games-auto-release-testing/201013-231404_ckjT <local_path> to copy everything.

This can be used to download artifacts for local testing/inspection. This requires the cloud sdk to be installed locally, to use the gsutil tool.

### CODE REVIEW

File: scripts/gha/gcs_uploader.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  if artifacts:
+    logging.info("Found the following artifacts:\n%s", "\n".join(artifacts))
+  else:
+    logging.info("No artifacts found to upload")
+    return
+
+  gcs.authorize_gcs(key_file_path)
+
+  gcs_prefix = "gs://%s/%s" % (gcs.PROJECT_ID, gcs.get_unique_gcs_id())
```
Comment (vimanyu) [inline scripts/gha/gcs_uploader.py:69]: Noticed this being used in test_lab.py too.
Maybe its better to make this a utility function too?

**Review by vimanyu — CHANGES_REQUESTED** [review]
Looks good for most part. 
Spotted a couple of places with code duplication where we could use the utility module.

File: scripts/gha/test_lab.py [hunk 2]
```diff
@@ -38,11 +38,7 @@
 
 """
 
-import datetime
```
Comment (DellaBitta) [inline scripts/gha/test_lab.py:41]: This file needs a copyright & license header as well.

**Review by DellaBitta — CHANGES_REQUESTED** [review]

**Comment by paulb777** [comment]
Consider adapting this presubmit test - https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/check_copyright.sh

**Comment by DellaBitta** [comment]
Yep, it's on our list of things to do!

File: .github/workflows/integration_tests.yml [hunk 3]
```diff
@@ -126,9 +126,13 @@ jobs:
       - name: Set CLOUDSDK_PYTHON (Windows)
         if: startsWith(matrix.os, 'windows') && !cancelled()
         run: echo "::set-env name=CLOUDSDK_PYTHON::${{env.pythonLocation}}\python.exe"
-      - name: Install Cloud SDK for mobile integration tests
-        if: matrix.target_platform != 'Desktop' && !cancelled()
+      - name: Install Cloud SDK
+        if: ${{ !cancelled() }}
```
[the remark this answers is not in the document]
Comment (anonymous-akorn) [reply .github/workflows/integration_tests.yml:130]: A quirk of Github's workflow syntax. The braces is the "full" syntax, and can usually be omitted in this context, but for some reason that was throwing an error in this case (the case of having nothing else in the condition).

File: scripts/gha/test_lab.py (hunk 2 above)
Comment (anonymous-akorn) [reply scripts/gha/test_lab.py:41]: Whoopsie

File: .github/workflows/integration_tests.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (anonymous-akorn) [reply .github/workflows/integration_tests.yml:130]: There are three statuses: success, failed, and cancelled. This ensures it runs on success or fail.

File: scripts/gha/gcs_uploader.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    testapp_dir (str): Prefix to strip from all paths.
+
+  Returns:
+    (str): GCS URI for this artifact.
+
+  """
+  return gcs_prefix + "/" + path[len(testapp_dir) + 1:].replace("\\", "/")
+
+
+def _fix_path(path):
```
[the remark this answers is not in the document]
Comment (anonymous-akorn) [reply scripts/gha/gcs_uploader.py:105]: I considered this, but couldn't think of a good name. "fixing" a path is relative to the use case, and isn't at all intuitive, which means someone would have to look up the utility module to figure out what this does. In some cases, one might want to also use os.path.realpath, but in most others, we don't. We could call it something like "abspath_and_expanduser", or add options for things like realpath, but in both cases I think it's more sensible to just use Python's tools directly here. I think this is also why Python doesn't have a function for this use case.

I also realize I may be massively overthinking this.

File: scripts/gha/gcs_uploader.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  Args:
+    gcs_prefix (str): Directory on GCS, of the form "gs://<id>/<etc>"
+    path (str): Full local path to artifact.
+    testapp_dir (str): Prefix to strip from all paths.
+
+  Returns:
+    (str): GCS URI for this artifact.
+
+  """
+  return gcs_prefix + "/" + path[len(testapp_dir) + 1:].replace("\\", "/")
```
[the remark this answers is not in the document]
Comment (anonymous-akorn) [reply scripts/gha/gcs_uploader.py:102]: Good call, the +1 in particular was bugging me about this code.

File: scripts/gha/gcs_uploader.py (hunk 1 above)
Comment (anonymous-akorn) [reply scripts/gha/gcs_uploader.py:69]: Yeah I basically already have a function that does this in test_lab, I can move it to gcs.

**Comment by anonymous-akorn** [comment]
Well that's new:

NDK_ROOT environment variable is not set, or NDK version is incorrect.
This build requires NDK r16b for STLPort compatibility, downloading...
+ curl -LSs https://dl.google.com/android/repository/android-ndk-r16b-windows-x86_64.zip --output /tmp/android-ndk-r16b.zip
curl: (16) Error in the HTTP2 framing layer
Error: Process completed with exit code 16.

**Review by vimanyu — APPROVED** [review]
LGTM

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**